pkg/util/helper/taint_test.go

```diff
@@ -357,3 +359,209 @@ func TestGetNoExecuteTaints(t *testing.T) {
+					},
+				},
+			},
+			want: -1,
+		},
+		{
+			name: "test no tolerations",
+			args: args{
+				noExecuteTaints: []corev1.Taint{
+					{
+						Key:       "key",
+						Value:     "value",
+						Effect:    "NoExecute",
+						TimeAdded: nil,
+					},
+				},
+				usedTolerations: nil,
+			},
+			want: 0,
+		},
+		{
+			name: "test get minimal toleration time",
+			args: args{
+				noExecuteTaints: []corev1.Taint{
+					{
+						Key:       "key",
+						Value:     "value",
+						Effect:    "NoExecute",
+						TimeAdded: nil,
+					},
+					{
+						Key:       "key",
+						Value:     "value",
+						Effect:    "NoExecute",
+						TimeAdded: &now,
+					},
+					{
+						Key:       "key",
+						Value:     "value",
+						Effect:    "NoExecute",
+						TimeAdded: &later,
+					},
+				},
+				usedTolerations: []corev1.Toleration{
+					{
+						Key:               "key",
+						Operator:          "Equal",
+						Value:             "value",
+						Effect:            "NoExecute",
+						TolerationSeconds: nil,
+					},
+					{
+						Key:               "key",
+						Operator:          "Equal",
+						Value:             "value",
+						Effect:            "NoExecute",
+						TolerationSeconds: pointer.Int64(10000),
+					},
+					{
+						Key:               "key",
+						Operator:          "Equal",
+						Value:             "value",
+						Effect:            "NoExecute",
+						TolerationSeconds: pointer.Int64(20000),
+					},
+				},
+			},
+			want: 9999999999999,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			if got := GetMinTolerationTime(tt.args.noExecuteTaints, tt.args.usedTolerations); got.Milliseconds() != tt.want.Milliseconds() {
+				t.Errorf("GetMinTolerationTime() = %v, want %v", got.Milliseconds(), tt.want.Milliseconds())
+			}
+		})
+	}
+}
+
+func TestGetMatchingTolerations(t *testing.T) {
+	type args struct {
+		taints      []corev1.Taint
+		tolerations []corev1.Toleration
+	}
+	tests := []struct {
+		name  string
+		args  args
+		want  bool
+		want1 []corev1.Toleration
+	}{
+		{
+			name: "test no taints",
+			args: args{
+				taints: nil,
+				tolerations: []corev1.Toleration{
+					{
+						Key:               "key",
+						Operator:          "Equal",
+						Value:             "value",
+						Effect:            "NoExecute",
+						TolerationSeconds: nil,
+					},
+					{
+						Key:               "key",
+						Operator:          "Equal",
+						Value:             "value",
+						Effect:            "NoExecute",
+						TolerationSeconds: pointer.Int64(10000),
+					},
+				},
+			},
+			want:  true,
+			want1: []corev1.Toleration{},
+		},
+		{
+			name: "test no tolerations",
+			args: args{
+				taints: []corev1.Taint{
+					{
+						Key:       "key",
+						Value:     "value",
+						Effect:    "NoExecute",
+						TimeAdded: nil,
+					},
+				},
+				tolerations: nil,
+			},
+			want:  false,
+			want1: []corev1.Toleration{},
+		},
+		{
+			name: "test return tolerations",
```

Comment:
How about changing the test case name to `don't exist mismatch taint`?

And Adding a case `exist mismatch taint` case will be better.

Reply:
ok